pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"sync"
+	"time"
+)
+
+// ServiceCAMetrics implements metrics gathering for this component.
+type ServiceCAMetrics struct {
+	lock       sync.Mutex
+	expiryTime prometheus.Gauge
+	timeLeft   prometheus.Gauge
+}
+
+func NewServiceCAMetrics() *ServiceCAMetrics {
+	return &ServiceCAMetrics{
+		expiryTime: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "service_ca_expiry_time",
+			Help: "Reports the expiry time of service CA used in the cluster.",
+		}),
+		timeLeft: prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "service_ca_time_left",
```

Comment:
this metric isn't really needed since it can be computed by Prometheus directly: `service_ca_expiry_time - time()`

Reply:
Thanks for the review. I removed the extra metric